src/zui/ZUIBadge/index.tsx

```diff
@@ -36,20 +44,30 @@ const useStyles = makeStyles<Theme, { status: ActivityStatus }>((theme) => ({
     height: '1rem',
     width: '1rem',
   },
+  text: {
+    fontFamily: theme.typography.fontFamily,
+    fontSize: '0.75rem',
+    fontWeight: 600,
+    lineHeight: '0.9rem',
+  },
 }));
 
-const ZUIBadge: FC<ZUIBadgeProps> = ({ number = undefined, status }) => {
-  const classes = useStyles({ status });
-  const style = number != undefined ? 'badge' : 'dot';
-  let displayValue = '';
+const ZUIBadge: FC<ZUIBadgeProps> = ({
+  number,
+  status,
+  truncateLargeNumber = false,
+}) => {
+  const classes = useStyles({ number, status });
 
-  if (number != undefined) {
-    displayValue = number <= 99 ? number.toString() : '99+';
-  }
+  const showValue = number != undefined;
+  const style = showValue ? 'badge' : 'dot';
 
   return (
     <Box className={classes[style]}>
-      <Typography variant="labelSmMedium">{displayValue}</Typography>
+      <Typography className={classes.text}>
+        {showValue && !truncateLargeNumber && number.toString()}
+        {showValue && truncateLargeNumber && number > 99 && '99+'}
```

Comment:
This logic will render nothing if `truncateLargeNumber` is true but the number is less than 99.

Reply:
dammit i was staring at this thinking i had prevented just that. thank you for noticing!